prime-router/src/main/kotlin/cli/tests/BasicTests.kt

```diff
@@ -178,6 +181,16 @@ class End2EndUniversalPipeline : CoolTest() {
                 if (!passed)
                     bad("***async end2end_up FAILED***: Send result invalid")
             }
+            val histories = sendResults.values.filterNotNull()
+            val reports = histories.flatMap { it.reports?.toList() ?: emptyList() }
+            val receiverNames = reports.map { "${it.receivingOrg}.${it.receivingOrgSvc}" }
+
+            expectedReceivers.forEach { receiver ->
```

Comment:
This is primary change... rather than relying on a hard coded number of receivers, the expectation is now to pass in the receivers that the test expects reports to have gone to